Xamarin.Essentials/Contacts/Contact.ios.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Contacts;
+using ContactsUI;
+using Foundation;
+using UIKit;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Contact
+    {
+        internal static Action<PhoneContact> CallBack { get; set; }
+
+        static Task<PhoneContact> PlatformPickContactAsync()
+        {
+            var uiView = Platform.GetCurrentViewController();
+            if (uiView == null)
+                throw new ArgumentNullException($"The View Controller can't be null.");
+
+            using var picker = new CNContactPickerViewController
+            {
+                Delegate = new ContactPickerDelegate()
+            };
+
+            uiView.PresentViewController(picker, true, null);
+            var source = new TaskCompletionSource<PhoneContact>();
+            try
+            {
+                CallBack = (phoneContact) =>
+                {
+                    var tcs = Interlocked.Exchange(ref source, null);
+                    tcs?.SetResult(phoneContact);
+                };
+            }
+            catch (Exception ex)
+            {
+                source.SetException(ex);
+            }
+            return source.Task;
+        }
+
+        internal static PhoneContact GetContact(CNContact contact)
+        {
+            if (contact == null)
+                return default;
+
+            try
+            {
+                var contactType = ToPhoneContact(contact.ContactType);
+                var phones = new Dictionary<string, ContactType>();
+
+                foreach (var item in contact.PhoneNumbers)
+                    phones.Add(item?.Value?.StringValue, contactType);
+
+                var emails = new Dictionary<string, ContactType>();
+
+                foreach (var item in contact.EmailAddresses)
+                    emails.Add(item?.Value?.ToString(), contactType);
```

Comment:
Same case as for numbers. I think a better approach would be to add a mapping function similar to the UWP implementation and use keys such as `CNLabelKey.Work` and `CNLabelKey.Home` to determine the e-mail address type. The iOS API has some constants defined, but users can also assign labels at will, and the Contacts app even suggests pre-defined labels that has no corresponding constant, so covering at least the key labels suggested by the OS would be smart.